### ISSUE TITLE
Username set to givenname for Okta social account

### ISSUE THREAD

**Issue by pdelpe** [pennersr/django-allauth#2956]
I have configured a new mailman-3 server with Okta as a social application. The authentication works well with Okta and the data comes through as expected. However, the username is being set as the given_name. 
Would it be possible to update the Okta provider to set the username correctly? It looks like the other social providers have this option set.

**Comment by anoushk1234** [comment]
@pdelpe Do want it set as the "name" or "nickname"
`json
{
  "sub": "00uid4BxXw6I6TV4m0g3",
  "name" :"John Doe",
  "nickname":"Jimmy",
  "given_name":"John",
  "middle_name":"James",
  "family_name":"Doe",
  "profile":"https://example.com/john.doe",
  "zoneinfo":"America/Los_Angeles",
  "locale":"en-US",
  "updated_at":1311280970,
  "email":"john.doe@example.com",
  "email_verified":true,
  "address" : { "street_address":"123 Hollywood Blvd.", "locality":"Los Angeles", "region":"CA", "postal_code":"90210", "country":"US" },
  "phone_number":"+1 (425) 555-1212"
}
`

**Comment by anoushk1234** [comment]
`def extract_common_fields(self, data):
        return dict(
            email=data["email"],
            name=data["name"],
            nickname=data["nickname"]
        )` does this work?

**Comment by pdelpe** [comment]
Hi @anoushk1234 , Thank you for your response. 

I have configured the Okta provider using the following settings,

SOCIALACCOUNT_PROVIDERS = {
    'okta': {
        'OKTA_BASE_URL': 'example.okta.com',
    }
}

The authentication works, however, the username is set as the first name (giveName)

[https://ibb.co/L5ht3zC](url)

**Comment by anoushk1234** [comment]
you can look at the json response i put above and accordingly choose what u want meanwhile ill submit this pr

**Comment by pennersr** [comment]
Currently, the code uses this:

```    def extract_common_fields(self, data):
        return dict(
            email=data["email"],
            last_name=data["family_name"],
            first_name=data["given_name"],
        )
```

That can be easily extended to add a `username=...`, but we need to know what attribute to pick -- is that `nickname` ?

**Comment by pennersr** [comment]
Fixed via 9535e153

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2956
